<?xml version='1.0' encoding='UTF-8' standalone='no'?>
<doxygen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="compound.xsd" version="1.8.17">
  <compounddef id="classdoxybook_1_1exception" kind="class" language="C++" prot="public">
    <compoundname>doxybook::exception</compoundname>
    <basecompoundref prot="public" virt="non-virtual">exception</basecompoundref>
    <includes refid="exception_8hpp" local="no">exception.hpp</includes>
      <sectiondef kind="private-attrib">
      <memberdef kind="variable" id="classdoxybook_1_1exception_1a4b571d99cdb2ed3c6e1acb3bae5c7a6d" prot="private" static="no" mutable="no">
        <type>std::string</type>
        <definition>std::string doxybook::exception::msg_</definition>
        <argsstring></argsstring>
        <name>msg_</name>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="doxybook/exception.hpp" line="34" column="21" bodyfile="doxybook/exception.hpp" bodystart="34" bodyend="-1"/>
      </memberdef>
      </sectiondef>
      <sectiondef kind="public-func">
      <memberdef kind="function" id="classdoxybook_1_1exception_1ac2182d669ee05215f976ec3132b772cc" prot="public" static="no" const="no" explicit="no" inline="no" virt="non-virtual">
        <type></type>
        <definition>doxybook::exception::exception</definition>
        <argsstring>()=default</argsstring>
        <name>exception</name>
        <briefdescription>
<para>Constructor. </para>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="doxybook/exception.hpp" line="22" column="9"/>
      </memberdef>
      <memberdef kind="function" id="classdoxybook_1_1exception_1a25536bef602853e6b5ac08deb4d34f95" prot="public" static="no" const="no" explicit="yes" inline="yes" virt="non-virtual">
        <type></type>
        <definition>doxybook::exception::exception</definition>
        <argsstring>(std::string msg)</argsstring>
        <name>exception</name>
        <param>
          <type>std::string</type>
          <declname>msg</declname>
        </param>
        <briefdescription>
<para>Constructor. </para>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="doxybook/exception.hpp" line="25" column="18" bodyfile="doxybook/exception.hpp" bodystart="25" bodyend="25"/>
      </memberdef>
      <memberdef kind="function" id="classdoxybook_1_1exception_1a6423e97a856fc1221ae73076249b6598" prot="public" static="no" const="yes" explicit="no" inline="yes" virt="non-virtual">
        <type>char const  *</type>
        <definition>char const* doxybook::exception::what</definition>
        <argsstring>() const override</argsstring>
        <name>what</name>
        <exceptions> throw ()</exceptions>
        <briefdescription>
<para>The error message. </para>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="doxybook/exception.hpp" line="29" column="9" bodyfile="doxybook/exception.hpp" bodystart="29" bodyend="31"/>
      </memberdef>
      </sectiondef>
    <briefdescription>
<para>Doxybook exceptions. </para>
    </briefdescription>
    <detaileddescription>
<para>This is a very basic class used to represent doxybook exceptions </para>
    </detaileddescription>
    <inheritancegraph>
      <node id="1">
        <label>doxybook::exception</label>
        <link refid="classdoxybook_1_1exception"/>
        <childnode refid="2" relation="public-inheritance">
        </childnode>
      </node>
      <node id="2">
        <label>std::exception</label>
      </node>
    </inheritancegraph>
    <collaborationgraph>
      <node id="1">
        <label>doxybook::exception</label>
        <link refid="classdoxybook_1_1exception"/>
        <childnode refid="2" relation="public-inheritance">
        </childnode>
      </node>
      <node id="2">
        <label>std::exception</label>
      </node>
    </collaborationgraph>
    <location file="doxybook/exception.hpp" line="19" column="42" bodyfile="doxybook/exception.hpp" bodystart="19" bodyend="35"/>
    <listofallmembers>
      <member refid="classdoxybook_1_1exception_1ac2182d669ee05215f976ec3132b772cc" prot="public" virt="non-virtual"><scope>doxybook::exception</scope><name>exception</name></member>
      <member refid="classdoxybook_1_1exception_1a25536bef602853e6b5ac08deb4d34f95" prot="public" virt="non-virtual"><scope>doxybook::exception</scope><name>exception</name></member>
      <member refid="classdoxybook_1_1exception_1a4b571d99cdb2ed3c6e1acb3bae5c7a6d" prot="private" virt="non-virtual"><scope>doxybook::exception</scope><name>msg_</name></member>
      <member refid="classdoxybook_1_1exception_1a6423e97a856fc1221ae73076249b6598" prot="public" virt="non-virtual"><scope>doxybook::exception</scope><name>what</name></member>
    </listofallmembers>
  </compounddef>
</doxygen>
